src/ksym.rs

```diff
@@ -91,29 +91,69 @@ impl KSymResolver {
         }
     }
 
-    pub fn find_address_ksym(&self, addr: u64) -> Option<&Ksym> {
+    pub fn find_address_ksym(&self, addr: u64) -> Vec<&Ksym> {
         let mut l = 0;
+        let mut dropped_l = 0;
         let mut r = self.syms.len();
 
         if !self.syms.is_empty() && self.syms[0].addr > addr {
-            return None;
+            return vec![];
         }
 
         while l < (r - 1) {
             let v = (l + r) / 2;
             let sym = &self.syms[v];
 
-            if sym.addr == addr {
-                return Some(sym);
-            }
             if addr < sym.addr {
                 r = v;
             } else {
+                if self.syms[l].addr != sym.addr {
+                    dropped_l = l;
+                }
                 l = v;
             }
         }
+        assert!(
+            l == 0 || self.syms[l].addr <= addr,
+            "left addr <= addr < right addr"
+        );
+        assert!(
+            r == self.syms.len() || addr < self.syms[r].addr,
+            "left addr <= addr < right addr"
+        );
```

Comment:
These are effectively assertions making claims about user provided data, correct? I don't think we should panic on potentially malformed input, but handle it gracefully, no?

Reply:
The assertions are making claims of the behavior of `KSymResolver`.  The symbols are sorted by `KSymResolver`.  These two assertions claim the facts that the later block depends on.